### ISSUE TITLE
make it possible to list template vdisks

### ISSUE THREAD

**Issue by zaibon** [zero-os/0-Disk#462]
Once we have the flow described in #460 in place, we'll need a way to list all the templates available in a storage cluster, this command should also works against the 0-hub to be able to list all templates available to imports.

**Comment by GlenDC** [comment]
I suppose that means we somehow store a list of vdisk ids, which were created using `zeroctl import template`?

**Comment by GlenDC** [comment]
Because we already have `zeroctl list vdisks`, but that lists all vdisks, there is no filter on templates, as templates are just a concept, it's not visible in the storage at all what is a template and what is not.

**Comment by zaibon** [comment]
Currently the way to create a templates is way to complex and cumbersome, so I think the creation of a template should also be handled by the `zeroctl`. Like that we can ensure that a certain format for the template is respected and list can then filter base on the name of the disk ?

**Comment by GlenDC** [comment]
@zaibon you are referring to creating them using `zeroctl import template`, no? Like I said, as long as they are created using `zeroctl import template`, and any other (new) zeroctl command in the future, we can maintain a set value in the ardb metadata server.

**Comment by zaibon** [comment]
> @zaibon you are referring to creating them using zeroctl import template

No I'm not. I'm referring to the creation of the template on the hub. See https://github.com/zero-os/home/blob/master/docs/tutorials/Create_a_vdisk_Boot_Image.md

The process described in this document should be reviewed and I think zeroctl should play a role there. Now it's all too manual, complex and cumbersome. No user would ever do that.

**Comment by GlenDC** [comment]
Anything the nbdserver does in those steps can easily be done using the zeroctl tool with some new kind of `zeroctl create template` command. However we would need to find a way to convert an image to to our storage blocks (which is now done using `qemu-img convert`). I don't think there is a standard Go solution available for that, so that might be quite a lot of work (to support such a image creation facility) if you want to automate all that into a zeroctl command...

**Comment by GlenDC** [comment]
@zaibon I created a seperate issue #463 for it. Which would provide a `zeroctl create template` for this purpose. For now I've put it under the Roadmap milestone as the beta-1 is already quite packed as it is. But if you really need it in beta-1, feel free to move it (but I can't promise it won't delay a release, as the beta-1 milestone is very packed as it is, so I wouldn't recommend doing so).

**Comment by GlenDC** [comment]
@zaibon in an earlier comment I proposed to add some new data (an ARDB list/set) which would collect the templates stored on an ARDB server. However I don't think this is a good idea because:

+ It would make all other commands which have to interact with that data also more complex (as it would need to ensure that data is also copied/deleted/moved/created/exported);
+ Templates are a high-level concept, outside the bounds of 0-Disk;

However, in all this time that 0-Disk is used by 0-Orchestrator I've noticed that the Cairo team already prefixes templates at all times with the prefix `template:`.

So here is what I'll do in a PR to resolve this issue:

1. Add an optional argument to the `ListVdisks` command (this can be `nil`):

```go
func ListVdisks(cluster ardb.StorageCluster, pred func(string) bool) ([]string, error)
```

When it's not `nil` it will used as an extra filter.

2. Add a optional flag `--name` which takes a Regexp String, which we can use to than use the [`(*Regexp).MatchString`](https://godoc.org/regexp#Regexp.MatchString) as the predicate mentioned above;

This solution allows you to list all templates and more. Without requiring to store any new data (which is a bad idea, as it would only making everything more error prone).
